extensions-core/lookups-cached-global/src/main/java/io/druid/query/lookup/namespace/ExtractionNamespaceCacheFactory.java

```diff
@@ -41,11 +44,32 @@
    * @param lastVersion         The version which was last cached
    * @param swap                The temporary Map into which data may be placed and will be "swapped" with the proper
    *                            namespace Map in NamespaceExtractionCacheManager. Implementations which cannot offer
-   *                            a swappable cache of the data may ignore this but must make sure `buildFn(...)` returns
+   *                            a swappable cache of the data may ignore this but must make sure `buildFnMap(...)` returns
    *                            a proper Function.
    *
    * @return A callable that will be used to refresh resources of the namespace and return the version string used in
    * the populating
    */
-  Callable<String> getCachePopulator(String id, T extractionNamespace, String lastVersion, Map<String, String> swap);
+  public Callable<String> getMapCachePopulator(
+      String id,
+      T extractionNamespace,
+      String lastVersion,
+      ConcurrentMap<String, Map<String, String>> swap,
+      Function<String, Map<String, String>> mapAllocator
+      )
+  {
+    String key = new NamespaceKey(id, KeyValueMap.DEFAULT_MAPNAME).getKey();
+    Map<String, String> cache = swap.get(key);
+    if (cache == null)
+    {
+      cache = mapAllocator.apply(key);
+      swap.put(key, cache);
+    }
+    return getCachePopulator(id, extractionNamespace, lastVersion, cache);
+  }
+
+  public Callable<String> getCachePopulator(String id, T extractionNamespace, String lastVersion, Map<String, String> swap)
```

Comment:
Can we add some javadocs explaining how this function differs from getMapCachePopulator()?